### PR TITLE
Add stubs for docutils.parsers.rst.directives.parts

### PR DESCRIPTION
**Pull request by adamtheturtle**

### CODE REVIEW

File: stubs/docutils/docutils/parsers/rst/directives/parts.pyi [hunk 1]
```diff
@@ -1,3 +1,10 @@
-from _typeshed import Incomplete
+from collections.abc import Sequence
 
-def __getattr__(name: str) -> Incomplete: ...
+from docutils.parsers.rst import Directive
+
+class Contents(Directive):
+    backlinks_values: Sequence[str]
```
Comment (srittau) [inline stubs/docutils/docutils/parsers/rst/directives/parts.pyi:6]:
```suggestion
    backlinks_values: Sequence[str]
    def backlinks(arg: str) -> str | None: ...
```

**Comment by adamtheturtle** [comment]
Thank you @srittau for taking a look at this.

I applied your suggested change, and then reverted it after CI failures:

```
stubs/docutils/docutils/parsers/rst/directives/parts.pyi:7: error: Self argument missing for a non-static method (or an invalid type for self)  [misc]
```

Do you have any suggestions for a way forward?

**Comment by srittau** [comment]
> Thank you @srittau for taking a look at this.
> 
> I applied your suggested change, and then reverted it after CI failures:
> 
> ```
> stubs/docutils/docutils/parsers/rst/directives/parts.pyi:7: error: Self argument missing for a non-static method (or an invalid type for self)  [misc]
> ```
> 
> Do you have any suggestions for a way forward?

Oops, I missed the `self` argument in that method.

**Comment by adamtheturtle** [comment]
> Oops, I missed the self argument in that method.

The source does not have a `self` argument:

https://github.com/docutils/docutils/blob/b768e2626088711dec257b0847b563d02700a712/docutils/docutils/parsers/rst/directives/parts.py#L31

```python
    def backlinks(arg):
        value = directives.choice(arg, Contents.backlinks_values)
        if value == 'none':
            return None
        else:
            return value
```

**Comment by srittau** [comment]
Ok, it's probably best to leave it out as you did originally.

**Review by srittau — APPROVED** [review]

**Pull request merged**